src/wallet/feebumper.cpp

```diff
@@ -150,12 +150,66 @@ bool TransactionCanBeBumped(const CWallet& wallet, const uint256& txid)
     if (wtx == nullptr) return false;
 
     std::vector<bilingual_str> errors_dummy;
-    feebumper::Result res = PreconditionChecks(wallet, *wtx, errors_dummy);
+    feebumper::Result res = PreconditionChecks(wallet, *wtx, /* require_mine=*/ true, errors_dummy);
     return res == feebumper::Result::OK;
 }
 
+struct SignatureWeights
+{
+private:
+    int m_sigs_count{0};
+    int64_t m_sigs_weight{0};
+
+public:
+    void AddSigWeight(const size_t weight, const SigVersion sigversion)
+    {
+        switch (sigversion) {
+        case SigVersion::BASE:
+            m_sigs_weight += weight * WITNESS_SCALE_FACTOR;
+            m_sigs_count += 1 * WITNESS_SCALE_FACTOR;
+            break;
+        case SigVersion::WITNESS_V0:
+            m_sigs_weight += weight;
+            m_sigs_count++;
+            break;
+        case SigVersion::TAPROOT:
+        case SigVersion::TAPSCRIPT:
+            assert(false);
```

Comment:
I think it might be good to catch all other calls to this function that are not covered by BASE or WITNESS_V0 as not being expected to be handled in this function. This will make this function future proof if other SigVersions get added in the future.

```suggestion
        case SigVersion::TAPSCRIPT:
        default:
            assert(false);
```

Reply:
It's actually better to not have a `default:` clause, as it means the compiler can warn about missing clauses, rather than needing to notice through an assertion failure.

Reply:
Yes, the lack of a `default` is intentional.

Reply:
Going to mark this as resolved.